midend/local_copyprop.cpp

```diff
@@ -35,10 +35,26 @@ static const IR::Expression *lvalue_out(const IR::Expression *exp) {
 
 /* helper function to extract the side effects from an expression as a statement */
 static const IR::Statement *makeSideEffectStatement(const IR::Expression *exp) {
-    while (auto unary = exp->to<IR::Operation_Unary>())
-        exp = unary->expr;
-    if (auto mc = exp->to<IR::MethodCallExpression>())
+    auto binary = exp->to<IR::Operation_Binary>();
+    auto unary = exp->to<IR::Operation_Unary>();
+
+    while (binary || unary) {
+        if (unary) {
+            exp = unary->expr;
+        }
+        if (binary) {
+            if (hasSideEffects(binary->left)) {
+                exp = binary->left;
+            } else if (hasSideEffects(binary->right)) {
```

Comment:
This is incorrect if both `->left` and `->right` have side effects (the side effects in `->right` will be lost).  Add a `BUG_CHECK(!hasSideEffects(binary->right)`... at line 47 at least.